packages/proxy/src/proxy.ts

```diff
@@ -714,19 +714,26 @@ export async function proxyV1({
                 }
 
                 if (delta.tool_calls) {
-                  if (!tool_calls) {
-                    tool_calls = [
+                  const lastTool = tool_calls
+                    ? tool_calls[tool_calls.length - 1]
+                    : undefined;
+                  if (
+                    !lastTool ||
+                    (lastTool.function &&
+                      delta.tool_calls[0].id &&
+                      lastTool.id !== delta.tool_calls[0].id)
+                  ) {
+                    tool_calls = (tool_calls ?? []).concat([
                       {
                         index: 0,
                         id: delta.tool_calls[0].id,
                         type: delta.tool_calls[0].type,
                         function: delta.tool_calls[0].function,
                       },
-                    ];
-                  } else if (tool_calls[0].function) {
-                    // TODO: what about parallel calls?
-                    tool_calls[0].function.arguments =
-                      (tool_calls[0].function.arguments ?? "") +
+                    ]);
+                  } else if (lastTool.function) {
```

Comment:
may want to see if this works:

```ts
if (delta.tool_calls?.length) {
  const lastTool = tool_calls?.[tool_calls?.length - 1];

  // do we need to add the tool call to our list?
  if (lastTool?.id !== delta.tool_calls?.[0]?.id) {
    tool_calls = (tool_calls ?? []).concat([
      {
        index: 0, // should this be tool_calls.length  ?
        id: delta.tool_calls[0].id,
        type: delta.tool_calls[0].type,
        function: delta.tool_calls[0].function,
      },
    ]);
    // same tool or no tool call
  } else if (lastTool?.function) {
    lastTool.function.arguments =
      (lastTool.function.arguments ?? "") +
      (delta.tool_calls[0].function?.arguments ?? "");
  }
}
```

Reply:
would be good to see a few tests on this (esp. locally)

Reply:
cleaned this up